SWLOR.Game.Server/Feature/ChatCommandDefinition/DMChatCommand.cs

```diff
@@ -406,10 +406,66 @@ private void SetLocalVariable()
 
                     SendMessageToPC(user, "Local string set: " + variableName + " = " + value);
                 });
+
+            _builder.Create("settag")
+                .Description("Sets a local tag on a target.")
+                .Permissions(AuthorizationLevel.DM, AuthorizationLevel.Admin)
+                .AvailableToAllOnTestEnvironment()
+                .RequiresTarget()
+                .Validate((user, args) =>
+                {
+                    if (args.Length <= 0)
+                    {
+                        return "Missing arguments. Format should be: /SetTag <VALUE>. Example: /SetTag DUNGEON_ENTRANCE";
+                    }
+
+                    return string.Empty;
+                })
+                .Action((user, target, location, args) =>
+                {
+                    if (!GetIsObjectValid(target))
```

Comment:
I'm on the fence about this one. Tags are used in some parts of the scripting and my concern is if this command is used on the wrong object we could run into systemic problems. Is there anything you can do to restrict its usage to the custom area transitions? Maybe abstract it away a bit more so that it's like:

/SetTransition MY_TRANSITION

and requiring the target object be a specific type of placeable?

SetTag is a little too exposed for my liking.

Reply:
I can require it so that it fails if the resref of the placeable doesn't match the Teleport Object placeable. That should sidestep any possible issues.